### PR TITLE
Fixed an issue where lines could be wrongfully removed

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: starcluster/node.py [hunk 1]
```diff
@@ -871,33 +871,44 @@ def add_to_etc_hosts(self, nodes):
...
+        with self.ssh.remote_file('/etc/hosts', 'w') as host_file:
+            print >> host_file, "\n".join(host_file_lines)
+            host_file.close()
+
+    @classmethod
+    def filter_etc_hosts_lines(cls, nodes, lines):
+        to_remove = \
+            [n.short_alias + "|" + n.private_ip_address.replace(".", "\.")
+             + "\s" for n in nodes]
+        print to_remove
```
Comment (jraby) [inline starcluster/node.py]: Should we keep that print?  or was it only used for testing/debugging the code?
Maybe we should keep it but use log() ?

File: starcluster/node.py [hunk 2]
```diff
@@ -871,33 +871,44 @@ def add_to_etc_hosts(self, nodes):
...
         /etc/hosts file
         """
-        aliases = [n.short_alias for n in nodes]
-        ips = [n.private_ip_address.replace(".", "\.") for n in nodes]
-        self.ssh.remove_lines_from_file('/etc/hosts',
-                                        '|'.join(aliases) + '|'.join(ips))
+        with self.ssh.remote_file('/etc/hosts', 'r') as host_file:
+            lines = host_file.read().split("\n")
+
+        lines, rejected = self.filter_etc_hosts_lines(nodes, lines)
```
Comment (jraby) [inline starcluster/node.py:907]: I'd add something like `log.debug("Removing entries from hosts file: %s", ",".join(rejected))`  or something along those lines. Would help if we ever need to track issues like this again.

**Comment by jraby** [comment]
+1 modulo comments.

I don't like messing with `/etc/hosts`, but this is more robust thatn what it replaces.

**Comment by FinchPowers** [comment]
Squash merged.

**Pull request closed without merging**